docs/en/ingest-management/integrations/package-signatures.asciidoc

```diff
@@ -1,4 +1,48 @@
-[role="exclude",id="package-signatures"]
+[[package-signatures]]
 = Package signatures
 
-coming[8.4.0]
+All integration packages published by Elastic have package signatures that
+prevent malicious attackers from tampering with package content. When you
+install an Elastic integration, {kib} downloads the package and verifies the
+package signature against a public key. If the package is unverified, you can
+choose to force install it. However, it's strongly recommended that you avoid
+installing unverified packages unless they are from trusted sources.
+
+[discrete]
+[[why-verify-packages]]
+== Why is package verification necessary?
+
+Integration packages contain instructions, such as ILM policies, transforms, and
+mappings, that can significantly modify the structure of your {es} indices.
+Relying solely on HTTPS DNS name validation to prove provenance of the package
+is not a safe practice. A determined attacker could forge a certificate and
+serve up packages intended to disrupt the target.
+
+Installing verified packages ensures that your integration software has not been
+corrupted or otherwise tampered with.
+
+[discrete]
+== Open questions for reviewers:
+
+- How do users verify Elastic packages when installing them through the API? Do
+they need to download Elastic's GPG key from
+https://artifacts.elastic.co/GPG-KEY-elasticsearch and store it on disk? What
+else?
+- What if a key has to be invalidated in the future? Do we need to document
```

Comment:
> What if a key has to be invalidated in the future?

If the elastic signing key changed, then existing package installs would still show as verified (as they were verified at the time of install) but they would then show as unverified when new packages are installed or existing ones are upgraded. You are right that the user can update the ocnfig (see answer below) to make this work again. 

Some more info on this here: https://github.com/elastic/elastic-package/issues/576#issuecomment-1197867649

> Would users provide the file path in kibana.yml (through the xpack.fleet.packageVerification.gpgKeyPath setting)?

Yes this is what they would have to do for existing deployments, the key would be updated on kibana upgrade as well.

>  Do we need to worry about documenting this now?

Maybe a small nod to the `xpack.fleet.packageVerification.gpgKeyPath` setting? Not sure!